docs/JLBP-0022.md

```diff
@@ -0,0 +1,56 @@
+---
+jlbp:
+  id: JLBP-22
+permalink: /JLBP-22
+---
+
+# Declare all dependencies
+
+If your code references a class—for example, by invoking a method in that class—declare a dependency that includes that class in your pom.xml, build.gradle,
+or equivalent. In Bazel this practice is called "strict deps".
+
+Code should not call methods, reference fields, or instantiate classes from _indirect_ dependencies. These are dependencies of the declared dependencies. Projects that rely on indirect dependencies have an annoying habit of breaking in unexpected ways when direct dependencies are upgraded. 
```

Comment:
Maybe there's something less judgmental to say instead of "have an annoying habit"? What about just "can break unexpectedly"?

Reply:
done